pkg/asset/cluster/tfvars.go

```diff
@@ -490,10 +490,15 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 
 		// Set this flag to use an existing folder specified in the install-config. Otherwise, create one.
 		preexistingFolder := installConfig.Config.Platform.VSphere.Folder != ""
-
+		vCenterURL := strings.SplitAfter(installConfig.Config.VSphere.VCenter, ":")
+		port := ""
+		if len(vCenterURL) > 1 && vCenterURL[1] != "443" {
+			port = vCenterURL[1]
+		}
```

Comment:
If `VSphere.VCenter` includes the port, which is now the working assumption, we no longer need to edit tfvars. When port was listed as a separate field, we needed to include port because ultimately we want to pass <VCenter>:<port> to tf. But since we are now including port in VCenter, no changes are necessary.

My suggestion about handling default port was also given with the first context and was really just a bit of a sugar. I would not worry about handling default port at all. If users provide default port along with their vcenter I would just treat it like any other valid port.